addons/wiki/models.py

```diff
@@ -160,7 +192,42 @@ def spam_check(self):
                 if isinstance(v, basestring)
             }
         user = OSFUser.load(user_id)
-        return self.wiki_page.node.check_spam(user, ['wiki_pages_latest'], request_headers)
+        return user, request_headers
+
+    def check_spam(self):
```

Comment:
Can we try something like:
```python

    def check_spam(self):
        user = get_current_user_()
	request = get_current_request()
        if not isinstance(request, DummyRequest):
            request_headers = {
                k: v
                for k, v in get_headers_from_request(request).items()
                if isinstance(v, basestring)
            }
        …
	…
```
And remove `_get_user_and_request_headers` from the model all together. It looks like `_get_user_and_request_headers` is only being called in check_spam. I just feel if `_get_user_and_request_headers` doesn’t have any wiki specific stuff in it, it shouldn’t have to be part of the Wiki model.

Reply:
okay, so I've removed _`get_user_and_request_headers` but am still using the `get_request_and_user_id` method inside `check_spam`.  I tried to use get_current_user() and get_current_request(), but I was getting all of these ` RuntimeError: working outside of request context` errors.  does this work for you?

Reply:
Yeah I get the same thing. I think the session object is out of scope in this situation. Anyway you solution works just as well.